### ISSUE TITLE
ObjectExpression doesn't work in variable declaration

### ISSUE THREAD

**Issue by mjethani** [eslint/eslint#8979]
**Tell us about your environment**

* **ESLint Version:** v4.2.0
* **Node Version:** v6.10.2
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**

Default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "env": {
    "browser": true,
    "es6": true
  },
  "extends": "eslint:recommended",
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "indent": [
      "error",
      2
    ],
    "indent-legacy": [
      "error",
      2,
      {"ObjectExpression": "first"}
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "double"
    ],
    "semi": [
      "error",
      "always"
    ],
    "no-unused-vars": [
      "off"
    ],
    "no-constant-condition": [
      "off"
    ]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
const {a,
       b,
       c} = null;
```

**What did you expect to happen?**

No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

```
/Users/mjethani/tmp/eslint/foo.js
  2:1  error  Expected indentation of 2 spaces but found 7  indent
  3:1  error  Expected indentation of 2 spaces but found 7  indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.
```

**Additional notes**

`VariableDeclarator` does not have a `"first"` option, but it would seem more natural for `ObjectExpression` to apply here.

**Comment by not-an-aardvark** [comment]
You don't seem to have `ObjectExpression` configured for the `indent` rule (it's only configured for `indent-legacy`).

That said, if you set `ObjectExpression: first` on the `indent` rule, it has still no effect in this case. That part has been fixed on master -- please wait for the [next release](https://github.com/eslint/eslint/issues/8921).

**Comment by mjethani** [comment]
Thanks, that was an error in the reporting. I definitely meant to report this for `indent` and not `indent-legacy`. Anyway, it's good to know that it's fixed.
